pkg/operator/ceph/object/controller.go

```diff
@@ -498,18 +496,21 @@ func (r *ReconcileCephObjectStore) startMonitoring(objectstore *cephv1.CephObjec
 	// Set the monitoring flag so we don't start more than one go routine
 	r.objectStoreChannels[objectstore.Name].monitoringRunning = true
 
-	var port string
+	var port int32
+	var sslenabled bool
 
 	if objectstore.Spec.Gateway.SecurePort != 0 && objectstore.Spec.Gateway.SSLCertificateRef != "" {
-		port = strconv.Itoa(int(objectstore.Spec.Gateway.SecurePort))
+		port = objectstore.Spec.Gateway.SecurePort
```

Comment:
What about a method in `objectstore.Spec` like `IsTLSEnabled` for that validation? Then you don't even need to pass it around and change the signature of `newBucketChecker()` since you can pass the entire spec.

Reply:
@leseb : Make sense, but I still need to change the signature of `newBucketchecker()` to include `objectstore` or `objectstore.Spec`. Am I missing something?

Reply:
Yes correct.